beacon_node/network/src/subnet_service/tests/mod.rs

```diff
@@ -338,110 +339,83 @@ mod attestation_service {
+        subnet_service
             .validator_subscriptions(vec![sub1, sub2].into_iter())
             .unwrap();
 
         // Unsubscription event should happen at slot 2 (since subnet id's are the same, unsubscription event should be at higher slot + 1)
-        // Get all events for 1 slot duration (unsubscription event should happen after 2 slot durations).
-        let events = get_events(&mut attestation_service, None, 1).await;
-        matches::assert_matches!(
-            events[..3],
-            [
-                SubnetServiceMessage::Subscribe(_any1),
-                SubnetServiceMessage::EnrAdd(_any3),
-                SubnetServiceMessage::DiscoverPeers(_),
-            ]
-        );
-
         let expected = SubnetServiceMessage::Subscribe(Subnet::Attestation(subnet_id1));
 
-        // Should be still subscribed to 2 long lived and up to 1 short lived subnet if both are
-        // different.
-        if !attestation_service.is_subscribed(
-            &subnet_id1,
-            attestation_subnets::SubscriptionKind::LongLived,
-        ) {
-            // The index is 3*subnets_per_node (because we subscribe + discover + enr per long lived
-            // subnet) + 1
-            let index = 3 * subnets_per_node;
-            assert_eq!(expected, events[index]);
-            assert_eq!(
-                attestation_service.subscription_count(),
-                subnets_per_node + 1
-            );
+        if subnet_service.is_subscribed(&Subnet::Attestation(subnet_id1)) {
+            // If we are permanently subscribed to this subnet, we won't see a subscribe message
+            let _ = get_events(&mut subnet_service, None, 1).await;
         } else {
-            assert!(attestation_service.subscription_count() == subnets_per_node);
+            let subscription = get_events(&mut subnet_service, None, 1).await;
+            assert_eq!(subscription, [expected]);
         }
 
         // Get event for 1 more slot duration, we should get the unsubscribe event now.
-        let unsubscribe_event = get_events(&mut attestation_service, None, 1).await;
+        let unsubscribe_event = get_events(&mut subnet_service, None, 1).await;
 
         // If the long lived and short lived subnets are different, we should get an unsubscription
         // event.
-        if !attestation_service.is_subscribed(
-            &subnet_id1,
-            attestation_subnets::SubscriptionKind::LongLived,
-        ) {
-            assert_eq!(
-                [SubnetServiceMessage::Unsubscribe(Subnet::Attestation(
-                    subnet_id1
-                ))],
-                unsubscribe_event[..]
-            );
+        let expected = SubnetServiceMessage::Unsubscribe(Subnet::Attestation(subnet_id1));
+        if !subnet_service.is_subscribed(&Subnet::Attestation(subnet_id1)) {
+            assert_eq!([expected], unsubscribe_event[..]);
         }
 
         // Should be subscribed 2 long lived subnet after unsubscription.
```

Comment:
I think this is checking:

```suggestion
        // Should no longer be subscribed to any short lived subnet after unsubscription.
```

Reply:
Agreed